Source/Scene/ShadowMap.js

```diff
@@ -150,69 +164,102 @@ define([
-        // TODO : Use nearest filtering for testing, not default linear.
-        shadowMap._depthStencilTexture = new Texture({
+    function createFramebufferColor(shadowMap, context) {
+        var depthRenderbuffer = new Renderbuffer({
             context : context,
             width : shadowMap._size,
             height : shadowMap._size,
-            pixelFormat : PixelFormat.DEPTH_STENCIL,
-            pixelDatatype : PixelDatatype.UNSIGNED_INT_24_8
+            format : RenderbufferFormat.DEPTH_COMPONENT16
+        });
+
+        var colorTexture = new Texture({
+            context : context,
+            width : shadowMap._size,
+            height : shadowMap._size,
+            pixelFormat : PixelFormat.RGBA,
+            pixelDatatype : PixelDatatype.UNSIGNED_BYTE,
+            sampler : createSampler()
+        });
+
+        var framebuffer = new Framebuffer({
+            context : context,
+            depthRenderbuffer : depthRenderbuffer,
+            colorTextures : [colorTexture]
         });
+
+        shadowMap._shadowMapTexture = colorTexture;
+        return framebuffer;
     }
 
-    function createFramebuffers(shadowMap, context) {
+    function createFramebufferDepth(shadowMap, context) {
+        var depthStencilTexture = new Texture({
+            context : context,
+            width : shadowMap._size,
+            height : shadowMap._size,
+            pixelFormat : PixelFormat.DEPTH_STENCIL,
+            pixelDatatype : PixelDatatype.UNSIGNED_INT_24_8,
+            sampler : createSampler()
+        });
+
         var framebuffer = new Framebuffer({
             context : context,
-            depthStencilTexture : shadowMap._depthStencilTexture,
-            destroyAttachments : false
+            depthStencilTexture : depthStencilTexture
         });
 
+        shadowMap._shadowMapTexture = depthStencilTexture;
+        return framebuffer;
+    }
+
+    function createFramebuffer(shadowMap, context) {
+        var createFunction = (context.depthTexture) ? createFramebufferDepth : createFramebufferColor;
+        var framebuffer = createFunction(shadowMap, context);
         shadowMap._framebuffer = framebuffer;
         shadowMap.passState.framebuffer = framebuffer;
         shadowMap.clearCommand.framebuffer = framebuffer;
     }
 
-    function updateFramebuffers(shadowMap, context) {
-        var depthStencilTexture = shadowMap._depthStencilTexture;
-        var textureChanged = !defined(depthStencilTexture) || (depthStencilTexture.width !== shadowMap._size);
-        if (!defined(shadowMap.framebuffer) || textureChanged) {
-            destroyTextures(shadowMap);
-            destroyFramebuffers(shadowMap);
-            createTextures(shadowMap, context);
-            createFramebuffers(shadowMap, context);
+    function updateFramebuffer(shadowMap, context) {
+        if (!defined(shadowMap.framebuffer) || (shadowMap._shadowMapTexture.width !== shadowMap._size)) {
```

Comment:
Is this supposed to be `shadowMap._framebuffer`? Right now, the framebuffer is being created and destroyed every frame.